### DISCUSSION TITLE
After Login the links wont work

### DISCUSSION THREAD

**Discussion by Fabianf1983** [kimai/kimai#4416] [Questions & Answers]
After try upgrading kimai to Version 2.2.1 with 2 days of problems.
It will start now but shown up the next problems. 
The most buttons wont work. If I click on the restart button it will load a new page instead the Dropdown I've seen bevore.
If I click on one of the Links ill receive an Error. 
`{"message":"Authentication required, missing user header: X-AUTH-USER"}`

If I click on Start, i can select the project but not the activity.
In Admin Mode, non of the dropdownbuttons will work.

If I use the Chrome DevTools -> Network
Everything looks like OK
![image](https://github.com/kimai/kimai/assets/143863941/6c063b8f-7b82-41ac-9baa-9b1f0d05ea2d)

Console shows some Errors
![image](https://github.com/kimai/kimai/assets/143863941/cdd46231-da84-4dae-8b7c-1c6722092273)

I configured a virtual Host for Apache as described
https://www.kimai.org/documentation/webserver-configuration.html#apache
`<virtualhost *:8080>
	ServerAdmin webmaster@test.localhost.com
	DocumentRoot "c:/xampp8/htdocs/kimai/public"
	ServerName kimai
	ServerAlias kimai
	ErrorLog "c:/xampp8/htdocs/kimai/var/log/kimai_error.log"
	CustomLog "c:/xampp8/htdocs/kimai/var/log/kimai_access.log" combined
	
	<Directory "C:/xampp8/htdocs/kimai/public">
        AllowOverride All

        # If you see an error like the following in your logs:
        # AH01630: client denied by server configuration: /var/www/kimai/public/
        # then you might have to exchange the "Order/Allow" rules with "Require" (see below)
        # More infos at https://httpd.apache.org/docs/2.4/de/upgrading.html

        # Apache 2.2
        #Order Allow,Deny
        #Allow from All
        
        # Apache 2.4
         Require all granted

        FallbackResource /index.php
    </Directory>
	<Directory "c:/xampp8/htdocs/kimai/public">
        Options FollowSymlinks
    </Directory>
	 <Directory "c:/xampp8/htdocs/kimai/public/bundles">
        FallbackResource disabled
    </Directory>
    
</virtualhost>`

I checked all the Plugins that needed by php and actived it.

I installed kimai into another folder of my webserver -> same problem
I tried different kinds of Browsers -> same problem
I deactived all Add- and Cookieblockers -> same problem

PHP Version: 8.2
Apache: 2.4

Sorry, but I dont what i do wrong...
It looks like a problem with the js files that shown up in the console.

Fabian

**Comment by Fabianf1983** [comment]
Still no Idea?

Kimaiwebloader wont load. There are some red Errors in the Google Chrome Console. Looks like a bug.

**Comment by Fabianf1983** [comment]
Restart server and set filepermissions to everyone wont work.
Also antivirus, firewall deactivated and tested from the server.
It keep still saying
_d a valid digest in the 'integrity' attribute for resource 'http://kimai:8080/build/app.9acfcfc8.js' with computed SHA-384 integrity 'ghF9wKKz/PkI1XqAfMwh6PYqUZ1FhhBdzjMYODuzZCdXdce1CNozceHr+a+OCMGQ'. The resource has been blocked._
and
_Uncaught ReferenceError: KimaiWebLoader is not defined_

**Comment by Fabianf1983** [comment]
Upgraded to 2.3.0 -> problem still there

**Comment by Fabianf1983** [comment]
figured out the Problem
just wondering y in my entrypoint.json line 66 is like this
'"/build/app.63b27bee.js": "sha384-1CV+he7m+KfetJpq/DmRwKfUhKH4L7MImv7rRUz/mr/KNR3VkT0fQJHF7ViPTVYE",'
instead
'"/build/app.63b27bee.js": "sha384-wwJlyhNjGeYv3Q0WhKHgUGLmKv9QTRphbk12zYoSWBDRCn5rdP0ZVYQm1xpjxB/p",'

**Reply by eduardogiorello** [reply]
tenía el mismo problema y soluciones con esto. No sé bien que es o que hace pero funciono. muchas gracias

**Comment by kevinpapst** [comment]
Your screenshot shows URLs that are NOT made Kimai.
Some browser plugin seems to interfer with your installation.
Maybe some caching or speed optimizer. 
Try another browser.

**Comment by mxgross** [comment]
I had the same issue with the 'KimaiWebloader is not defined'.

Executing
`yarn install`

And rebuilding all assets:
`yarn run prod`
did solve it.
Like described here: [https://www.kimai.org/documentation/developers.html#frontend-dependencies](https://www.kimai.org/documentation/developers.html#frontend-dependencies)

**Reply by kevinpapst** [reply]
NO. Please everyone. Don't! 

Kimai is installed hundreds of times every day and it works.
If your specific installation doesn't work, then something is fishy. 
Either with your setup or your environment.

1. Make sure you are not running in a subdirectory and your webroot points to kimai/public/
2. Run `bin/console assets:install` 

If both can't solve the problem, it might be:
- a browser plugin
- or a network service that rewrites URLs for caching, e.g. Cloudflare has some tools that do so

There are other possible reasons, like an update without proper executed install routine.

**Comment by akuehnis** [comment]
A fresh install of Kimai 2.27.0 with PHP8.3.15, following the 'Setup with SSH'. Problems in all Browsers, there are definitively no browser plugins installed. Running with apache 2.4.52 on Ubuntu 22.04 (App in Windows 11).
I do not have yarn installed.

My quick&dirty solution: 
* open public/build/entrypoints.json
* delete the integrity section completely and save
* `php bin/console cache:clear`

 No more integrity checks - but it works.

**Reply by kevinpapst** [reply]
So instead of fixing the problem, you are manipulating the software? 
Turn off your Cloudflare (or other proxy) "web optimization" or caching features and it will work.
